### DISCUSSION TITLE
How to deploy from a `mtar_url` which is in a private github repo?

### DISCUSSION THREAD

**Discussion by codeyogi911** [SAP/terraform-provider-cloudfoundry#178] [General]
I am trying to deploy mtar which is uploaded to release assets of a private github repo. What is the way to authenticate?

**Comment by Dray56** [comment]
Hi @codeyogi911 ,

The MTA controller expects a publicly accessible endpoint for the URL else it fails with 403 and there is no way to embed any additional authentication details for the request. Since the MTA controller client is used underneath the provider, this is not possible via terraform and cannot be implemented. Request you to kindly try from a public endpoint or authenticate and download them locally and upload via `mtar_path`.
